cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -1668,5 +1706,36 @@ pub fn define(
         }
     }
 
+    // Reference type instructions
+
+    // Null references implemented as iconst 0
+    e.enc32(null.bind_ref(R32), rec_pu_id_ref.opcodes(vec![0xb8]));
+
+    e.enc64(null.bind_ref(R32), rec_pu_id_ref.rex().opcodes(vec![0xb8]));
+    e.enc64(null.bind_ref(R32), rec_pu_id_ref.opcodes(vec![0xb8]));
```

Comment:
Hmm, do we want to support references of 32-bits on a 64-bit platform, in general? If so, I think the stack map records should contain the size of the reference value too. (cc @sunfishcode)

Reply:
Maybe we shouldn't? We use a 1 bit = 1 word mapping when we create the `Stackmap` metadata. This is an implicit assumption that references are always 1 word long. If we choose to support 32-bits references, we would need to make them not word-aligned.

Reply:
Talked to @sunfishcode in person. We decided to remove the lines that support 32-bit references on 64-bit platform. It's easier to add them back later if we wish. For now, we don't want that to be a part of this change.